Providers/Scripts/2.4x-2.5x/Scripts/nxOMSPlugin.py

```diff
@@ -277,13 +321,18 @@ def get_immediate_subdirectories(a_dir):
         return []
 
             
-def copy_all_files(src, dest):
+def copy_all_files(src, dest, is_exec):
     try:
         src_files = os.listdir(src)
         for file_name in src_files:
-            full_file_name = os.path.join(src, file_name)
-            if os.path.isfile(full_file_name):
-                shutil.copy(full_file_name, dest)
+            full_src_file = os.path.join(src, file_name)
+            full_dest_file = os.path.join(dest, file_name)
+            if os.path.isfile(full_src_file):
+                shutil.copy(full_src_file, dest)
+                if is_exec:
+                    mode = os.stat(full_dest_file).st_mode
+                    mode |= 0555
```

Comment:
Does it need to be executable by every user? I would recommend adding 0550 permissions, maybe even 0500, but not more.